WebContent/tetsgame.js

```diff
@@ -1,123 +1,188 @@
-var gridheight = 10;
-var gridwidth = 6;
-
-/* create grid of 0's */			
-function creategrid() {
-	var grid =[];
-	for (i = 0;  i< gridheight; i++) {
-		grid.push([]);
-		for (j=0; j< gridwidth; j++) {
-			grid[i].push(0);
+/**
+ * 
+ */
+
+var NewGrid = function(height, width) {
+	this.height = height;
+	this.width = width;
+	this.rows = [];
+	for (var i = 0;  i< this.height; i++) {
+		var row = [];
+		for (var j = 0; j< this.width; j++) {
+			row.push(0);
 		}
+		this.rows.push(row);
 	}
-	return grid
-}
-var grid = creategrid();
+	
+	this.addCoordinate = function(coordinates,colour) {
+		coordinates.forEach(function(coordinate) {
+            this.rows[coordinate[1]][coordinate[0]] = colour;
+        }, this);
+    }
+	
+	this.willFit = function(shiftedCoordinates) {
+		for(var r=0; r<shiftedCoordinates.length; r++){
+			if (shiftedCoordinates[r][0]<=-1){
+				return false;
+			}
+			if(shiftedCoordinates[r][0]>= this.width){
+				return false;
+			}
+			if (shiftedCoordinates[r][1]>= this.height){
+				return false;
+			} else {			// check grid for none 0 elements
+				x=shiftedCoordinates[r][0];
+				y=shiftedCoordinates[r][1];
+					if (this.rows[y][x] != 0){
+						console.log("collision");
+						return false;
+					}
+			} 
+		} return true; 
+	}
+};
 
+var grid = new NewGrid(22,10);
 
+ /* function draw(grid) {
+	var gridString = grid.rows.map(function(row) {
+       return row.join(" "); 
+    }).join("<br/>");
+    $("div").html(gridString);
+}
 
+draw(grid); */
 
-/* creating game board */
-var canvas =document.getElementById('tetrisgame');
+// creating game board
+var canvas = document.getElementById('tetrisgame');
 var context = canvas.getContext("2d");
-var bblock = new Image();
-bblock.src = 'bblock.gif';
-
-var colblock = new Image();
-colblock.src = 'darkblue.gif';
- 
 
-function gameboard() {
-for (i= 0;  i< gridheight; i++) {
-	for (j=0; j< gridwidth; j++) {
-		if (grid[i][j] == 0 ){ 
-			context.drawImage(bblock, j*20, i*20);
+function gameboard() {		// can't seem to figure out how to save game board once reaching the bottom.
+for (var x=0;  x< grid.height-2; x++) {
+	for (var col=0; col< grid.width; col++) { // copy gameboard() at point of collision.
+		if (grid.rows[x+2][col] == 0 ){
+			context.beginPath();
+			context.fillStyle = "grey"; 
+			context.rect(col*20, x*20, 20, 20);  
+			context.fill();
 		} else {
-			context.drawImage(colblock, j*20, i*20);
-		}
-	}
-			
-	}
+			context.beginPath();					// draws block onto grid
+			context.fillStyle = grid.rows[x+2][col]; 
+			context.rect(col*20, x*20, 20, 20);  
+			context.fill(); 
+	} 			
+}
+}
 }
 
+gameboard(); 
```

Comment:
Maybe this should rename ```drawGrid(grid)```, and similarly we could rename ```addBlock``` to ```drawBlock(block)```. The we could even have a function ```draw(grid, block)``` which calls each of them to draw the whole thing.